Add one more generic type for TypeScript users

dusansimic opened the pull request:
Adding another generic type to the functions allows developer to easily
specify the data type he wants to receive and return to the emitter.

Closes: https://github.com/sindresorhus/electron-better-ipc/issues/23

sindresorhus commented on index.d.ts:
````diff
@@ -25,11 +25,11 @@ export interface MainProcessIpc extends IpcMain {
 	```
 	*/
-	callRenderer<T>(
+	callRenderer<T, S>(
````
I think we should use more descriptive type names than `T` and `S` now that there are multiple.

sindresorhus commented:
Can you also add support for https://github.com/sindresorhus/electron-better-ipc/commit/3112e7be83eb03e6359beb83fce10f2fc0be8286 ?

sindresorhus commented:
Would be good to have a few type tests for this: https://github.com/sindresorhus/electron-better-ipc/blob/master/index.test-d.ts

dusansimic commented:
Added all requested changes. Does everything look ok?

dusansimic commented:
I found a bug while working on the issue mentioned in the description of PR and now fixed it with this small change. The problem was in the type of arguments passed to `callRenderer` and type of parameters of `answerMain`. The thing is TypeScript automatically adds a `undefined` type to a parameter if its specified as optional so that made a problem with `callRenderer` and `answerMain` functions as there was an error that argument and parameter types were not compatible as shown in the image.
![image](https://user-images.githubusercontent.com/12685466/71426766-36475c00-26af-11ea-8abd-8b709bcc9865.png)
This problem is now fixed and everything is ready for review.

UberMouse commented:
Any news on this? I've switched to pulling down directly off @dusansimic fork so I can get access to the typings. Would love to be able to switch back to the normal NPM package
